src/storage/v3/shard_rsm.cpp

```diff
@@ -112,6 +124,248 @@ Value ConstructValueVertex(const memgraph::storage::v3::VertexAccessor &acc, mem
+  std::vector<PropertyId> edge_props;
+  std::vector<memgraph::storage::v3::LabelId> labels_in_shard;
+
+  if (v_acc) {
+    const auto labels = v_acc->Labels(View::OLD);
+    labels_in_shard.reserve(labels.GetValue().size());
+
+    for (const auto &label : labels.GetValue()) {
+      labels_in_shard.emplace_back(label);
+    }
+
+    vertex_props = GetPropertiesFromAcessor(v_acc->Properties(View::OLD).GetValue());
+  }
+  if (e_acc) {
+    edge_props = GetPropertiesFromAcessor(e_acc->Properties(View::OLD).GetValue());
+  }
+
+  std::vector<PropertyId> all_properties(vertex_props.size() + edge_props.size());
+
+  // Sorting might not needed, since the props are being converted from std::map
+  std::sort(vertex_props.begin(), vertex_props.end());
+  std::sort(edge_props.begin(), edge_props.end());
+
+  auto it = std::set_symmetric_difference(vertex_props.begin(), vertex_props.end(), edge_props.begin(),
+                                          edge_props.end(), all_properties.begin());
+  all_properties.resize(it - all_properties.begin());
+
+  ctx.properties = std::move(all_properties);
+  ctx.labels = std::move(labels_in_shard);
+
+  return ctx;
+}
+
+std::any ParseExpression(const std::string &expr, memgraph::expr::AstStorage &storage) {
+  memgraph::frontend::opencypher::Parser<memgraph::frontend::opencypher::ParserOpTag::EXPRESSION> parser(expr);
+  memgraph::expr::ParsingContext pc;
+  memgraph::expr::CypherMainVisitor visitor(pc, &storage);
+
+  auto *ast = parser.tree();
+  return visitor.visit(ast);
+}
+
+memgraph::expr::TypedValue ComputeExpression(memgraph::expr::DbAccessor &dba,
+                                             const std::optional<memgraph::storage::v3::VertexAccessor> &v_acc,
+                                             const std::optional<memgraph::storage::v3::EdgeAccessor> &e_acc,
+                                             const std::string &expression, std::string_view node_name,
+                                             std::string_view edge_name) {
+  memgraph::expr::AstStorage storage;
+  memgraph::expr::Frame<memgraph::expr::TypedValue> frame{static_cast<int64_t>(128)};
+  memgraph::expr::SymbolTable symbol_table;
+
+  auto ctx = CreateEvaluationContext(v_acc, e_acc);
+  memgraph::storage::v3::ExpressionEvaluator eval{&frame, symbol_table, ctx, &dba, memgraph::expr::View::OLD};
+
+  auto expr = ParseExpression(expression, storage);
+
+  auto node_identifier = memgraph::expr::Identifier(std::string(node_name), false);
+  bool is_node_identifier_present = false;
+  auto edge_identifier = memgraph::expr::Identifier(std::string(edge_name), false);
+  bool is_edge_identifier_present = false;
+
+  std::vector<memgraph::expr::Identifier *> identifiers;
+
+  if (v_acc && expression.find(node_name) != std::string::npos) {
+    is_node_identifier_present = true;
+    identifiers.push_back(&node_identifier);
+  }
+  if (e_acc && expression.find(node_name) != std::string::npos) {
```

Comment:
Isn't it edge_name instead of node_name?

Reply:
Yes, that should be edge_name. Good catch.